documentation/prometheus-mixin/alerts.libsonnet

```diff
@@ -0,0 +1,176 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'prometheus',
+        rules: [
+          {
+            alert: 'PrometheusBadConfig',
+            expr: |||
+              prometheus_config_last_reload_successful{%(prometheusSelector)s} == 0
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              message: 'Prometheus failed to reload config, see container logs',
+            },
+          },
+          {
+            alert: 'PrometheusNotificationQueueRunningFull',
+            expr: |||
+              predict_linear(prometheus_notifications_queue_length{%(prometheusSelector)s}[5m], 60 * 30)
+              >
+              prometheus_notifications_queue_capacity{%(prometheusSelector)s}
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              message: "Prometheus' alert notification queue is running full for {{$labels.namespace}}/{{ $labels.pod}}",
```

Comment:
This doesn't describe the alert

Also, summary and description are what we use.

Reply:
> summary and description are what we use.

This isn't something we've agreed on - the duplication between summary and description is particularly annoying.  Lets take this to the list.

Reply:
It is, there was a big discussion on it a few years back.

Reply:
The outcome of the discussion a few years back was that setups are so different that we don't want to mandate anything and thus made the annotations completely free-form. At SoundCloud, we went into the habit of making `summary` something fairly static so that it would not vary between different alerts ending up in the same notification group, while `description` was tailored to the individual alert, describing in human-readable form what was going on. In that way, you could use `summary` as the headline for a notification group (e.g. mail subject) and then iterate through individual alerts e.g. in the mail text:
```
DiskRunningFull [2 alerts firing]:
- 98% full on foo.
- 95% full on bar.
```

If that's what everybody could live with, I'd apply it here.

Reply:
As the kube-prometheus alert mixins are using summary/description too (with approximately the semantics I have explained above), I go with them for now.